src/streaming/StreamProcessor.js

```diff
@@ -333,14 +437,155 @@ function StreamProcessor(config) {
+            return null;
+        }
+        /**
+         * This is critical for IE/Safari/EDGE
+         * */
         if (bufferController) {
-            bufferController.switchInitData(streamInfo.id, representationId, bufferResetEnabled);
+            let range = bufferController.getRangeAt(time);
+            const playingRange = bufferController.getRangeAt(currentTime);
+            if ((range !== null || playingRange !== null) && !hasSeekTarget) {
+                if (!range || (playingRange && playingRange.start != range.start && playingRange.end != range.end)) {
+                    const hasDiscontinuities = bufferController.getBuffer().hasDiscontinuitiesAfter(currentTime);
+                    if (hasDiscontinuities && getType() !== Constants.FRAGMENTED_TEXT) {
+                        fragmentModel.removeExecutedRequestsAfterTime(playingRange.end);
+                        bufferIsDivided = true;
+                    }
+                }
+            }
         }
+
+        if (requestToReplace) {
+            time = requestToReplace.startTime + (requestToReplace.duration / 2);
+            request = getFragmentRequest(representationInfo, time, {
+                timeThreshold: 0,
+                ignoreIsFinished: true
+            });
+        } else {
+            // Use time just whenever is strictly needed
+            request = getFragmentRequest(representationInfo,
+                hasSeekTarget || bufferIsDivided ? time : undefined, {
+                keepIdx: !hasSeekTarget && !bufferIsDivided
+            });
+
+            // Then, check if this request was downloaded or not
+            while (request && request.action !== FragmentRequest.ACTION_COMPLETE && fragmentModel.isFragmentLoaded(request)) {
+                // loop until we found not loaded fragment, or no fragment
+                request = getFragmentRequest(representationInfo);
+            }
+        }
+
+        return request;
+    }
+
+    function onTimedTextRequested(e) {
+        if (e.streamId !== streamInfo.id) return;
+
+        //if subtitles are disabled, do not download subtitles file.
+        if (textController.isTextEnabled()) {
+            const representation = representationController ? representationController.getRepresentationForQuality(e.index) : null;
+            const request = indexHandler ? indexHandler.getInitRequest(getMediaInfo(), representation) : null;
+            scheduleController.processInitRequest(request);
+        }
+    }
+
+    function onMediaFragmentLoaded(e) {
+        const chunk = e.chunk;
+        if (chunk.streamId !== streamInfo.id || chunk.mediaInfo.type != type) return;
+
+        const bytes = chunk.bytes;
+        const quality = chunk.quality;
+        const currentRepresentation = getRepresentationInfo(quality);
+
+        // Update current representation info (to update fragmentDuration for example in case of SegmentTimeline)
+        scheduleController.setCurrentRepresentation(currentRepresentation);
```

Comment:
Question: Is it possible that the fragmentDuration changed at this point? Why does the representation info change when a fragment has been loaded?

Reply:
Yes, for SegmentTimeline the segmentDuration (and thus fragmentDuration) is set and available only once a fragment has been requested (in TimelineSegmentsGetter)